probe/resolver.go

```diff
@@ -92,13 +92,15 @@ func (r staticResolver) resolveHosts() {
 			}
 		}
 
+		endpoints := make([]string, 0, len(addrs))
 		for _, addr := range addrs {
 			// For now, ignore IPv6
 			if addr.To4() == nil {
 				continue
 			}
-			r.add(net.JoinHostPort(addr.String(), peer.port))
+			endpoints = append(endpoints, net.JoinHostPort(addr.String(), peer.port))
 		}
+		r.set(peer.hostname, endpoints)
```

Comment:
Actually looks like set never actually removes existing ones, do above comment is incorrect.

Its not clear to me the behaviour of the code when a app restarts (and its id changes).  Could you elaborate please?